File: frontend/src/api/feishu.ts
import { request } from './request'

export interface FeishuHealthResponse {
  status: string
  configured?: boolean
  app_id?: boolean
  app_secret?: boolean
  base_url?: string
}

export interface FeishuBitablePreviewRequest {
  app_token: string
  table_ids?: string[]
  sample_size?: number
}

export interface FeishuBitableSyncTarget {
  target_table: 'industry_stats' | 'company_data' | 'policy_data'
  table_id: string
  field_mapping: Record<string, string>
  match_fields?: string[]
  clear_before_sync?: boolean
  sample_limit?: number
}

export interface FeishuBitableSyncRequest {
  app_token: string
  targets: FeishuBitableSyncTarget[]
  sample_only?: boolean
}

export interface FeishuDocPublishRequest {
  title: string
  result: Record<string, unknown>
  folder_token?: string
}

export interface FeishuIMSendRequest {
  receive_id: string
  receive_id_type?: string
  message: string
  uuid?: string
}

export function getHealth() {
  return request.get<FeishuHealthResponse>('/feishu/health', { loading: false })
}

export function getTokenStatus() {
  return request.get<{ status: string; has_token: boolean; expires_cached: boolean }>('/feishu/token', { loading: false })
}

export function previewBitable(params: FeishuBitablePreviewRequest) {
  return request.post<Record<string, unknown>>('/feishu/bitable/preview', params, { loading: false })
}

export function listTables(appToken: string) {
  return request.get<Record<string, unknown>>(`/feishu/bitable/${appToken}/tables`, { loading: false })
}

export function listRecords(appToken: string, tableId: string, pageSize: number = 500) {
  return request.get<Record<string, unknown>>(
    `/feishu/bitable/${appToken}/tables/${tableId}/records`,
    { params: { page_size: pageSize }, loading: false },
  )
}

export function syncBitable(params: FeishuBitableSyncRequest) {
  return request.post<Record<string, unknown>>('/feishu/bitable/sync', params, { loading: false })
}

export function publishDoc(params: FeishuDocPublishRequest) {
  return request.post<Record<string, unknown>>('/feishu/doc/publish', params, { loading: false })
}

export function sendIM(params: FeishuIMSendRequest) {
  return request.post<Record<string, unknown>>('/feishu/im/send', params, { loading: false })
}

File: backend/app/schemas/feishu.py
from typing import Any, Dict, List, Optional, Literal

from pydantic import BaseModel, Field


class FeishuTableSyncTarget(BaseModel):
    target_table: Literal["industry_stats", "company_data", "policy_data"]
    table_id: str = Field(..., description="飞书多维表格 table_id")
    field_mapping: Dict[str, str] = Field(default_factory=dict, description="飞书字段 -> 本地模型字段 映射")
    match_fields: Optional[List[str]] = Field(None, description="本地 upsert 匹配字段")
    clear_before_sync: bool = Field(False, description="同步前是否清空目标表")
    sample_limit: int = Field(500, ge=1, le=500, description="每个表最多同步/预览条数")


class FeishuBitablePreviewRequest(BaseModel):
    app_token: str = Field(..., description="飞书多维表格 app_token")
    table_ids: Optional[List[str]] = Field(None, description="可选，只预览指定表")
    sample_size: int = Field(3, ge=1, le=10, description="预览样本数")


class FeishuBitableSyncRequest(BaseModel):
    app_token: str = Field(..., description="飞书多维表格 app_token")
    targets: List[FeishuTableSyncTarget] = Field(..., min_length=1, description="同步目标配置")
    sample_only: bool = Field(False, description="仅预览不写库")


class FeishuDocPublishRequest(BaseModel):
    title: str = Field(..., description="飞书文档标题")
    result: Dict[str, Any] = Field(..., description="分析结果 payload")
    folder_token: Optional[str] = Field(None, description="飞书文档目录 token，可选")


class FeishuIMSendRequest(BaseModel):
    receive_id: str = Field(..., description="接收方 ID，群聊时为 chat_id")
    receive_id_type: str = Field("chat_id", description="接收方类型，默认 chat_id")
    message: str = Field(..., description="要发送的文本内容")
    uuid: Optional[str] = Field(None, description="消息幂等 ID，可选")
